Fix unused variable warnings when building in release mode

```
--- stderr: email                                                                                                                                                        
/home/chris/rmw_email/ws/src/rmw_email/email/src/wait.cpp: In function ‘std::pair<std::__cxx11::basic_string<char>, email::MessageInfo> email::wait_for_message_with_info(email::Subscription*, std::chrono::milliseconds)’:
/home/chris/rmw_email/ws/src/rmw_email/email/src/wait.cpp:39:14: warning: unused variable ‘timedout’ [-Wunused-variable]
   39 |   const bool timedout = waitset.wait(timeout);
      |              ^~~~~~~~
/home/chris/rmw_email/ws/src/rmw_email/email/src/wait.cpp: In function ‘std::pair<std::__cxx11::basic_string<char>, email::ServiceInfo> email::wait_for_response_with_info(email::SequenceNumber, email::ServiceClient*, std::chrono::milliseconds)’:
/home/chris/rmw_email/ws/src/rmw_email/email/src/wait.cpp:77:14: warning: unused variable ‘timedout’ [-Wunused-variable]
   77 |   const bool timedout = waitset.wait(timeout);
      |              ^~~~~~~~
/home/chris/rmw_email/ws/src/rmw_email/email/src/wait.cpp: In function ‘std::pair<email::ServiceRequest, email::ServiceInfo> email::wait_for_request_with_info(email::ServiceServer*, std::chrono::milliseconds)’:
/home/chris/rmw_email/ws/src/rmw_email/email/src/wait.cpp:117:14: warning: unused variable ‘timedout’ [-Wunused-variable]
  117 |   const bool timedout = waitset.wait(timeout);
      |              ^~~~~~~~
---
Finished <<< email [3min 33s]

--- stderr: rmw_email_cpp                                                                                                                                 
LTTng found: tracing enabled
/home/chris/rmw_email/ws/src/rmw_email/rmw_email_cpp/src/rmw_publish.cpp: In function ‘rmw_ret_t rmw_publish_serialized_message(const rmw_publisher_t*, const rmw_serialized_message_t*, rmw_publisher_allocation_t*)’:
/home/chris/rmw_email/ws/src/rmw_email/rmw_email_cpp/src/rmw_publish.cpp:79:8: warning: unused variable ‘pub’ [-Wunused-variable]
   79 |   auto pub = static_cast<email::Publisher *>(publisher->data);
      |        ^~~
---
Finished <<< rmw_email_cpp [59.6s]
```

--- a/email/src/wait.cpp
+++ b/email/src/wait.cpp
@@ -38,6 +38,7 @@ wait_for_message_with_info(
   waitset.add_subscription(subscription);
   const bool timedout = waitset.wait(timeout);
   assert(!timedout);
+  static_cast<void>(timedout);
 
   return subscription->get_message_with_info().value();
 }
@@ -76,6 +77,7 @@ wait_for_response_with_info(
   waitset.add_client(client);
   const bool timedout = waitset.wait(timeout);
   assert(!timedout);
+  static_cast<void>(timedout);
 
   return client->get_response_with_info(sequence_number).value();
 }
@@ -116,6 +118,7 @@ wait_for_request_with_info(
   waitset.add_server(server);
   const bool timedout = waitset.wait(timeout);
   assert(!timedout);
+  static_cast<void>(timedout);
 
   return server->get_request_with_info().value();
 }

--- a/rmw_email_cpp/src/rmw_publish.cpp
+++ b/rmw_email_cpp/src/rmw_publish.cpp
@@ -78,6 +78,7 @@ extern "C" rmw_ret_t rmw_publish_serialized_message(
 
   auto pub = static_cast<email::Publisher *>(publisher->data);
   assert(pub);
+  static_cast<void>(pub);
 
   RMW_SET_ERROR_MSG("rmw_publish_serialized_message not implemented for rmw_email_cpp");
   return RMW_RET_UNSUPPORTED;

